kube/objects.go

```diff
@@ -268,6 +268,39 @@ func objectsWithoutNils(objects *Objects) *Objects {
 	return objects
 }
 
+func NewClientConfigFromKubeConfig(opts *options) (*rest.Config, error) {
+	config := &rest.Config{}
+
+	err := opts.validate()
```

Comment:
Move opts.validate to NewClient and add another validation in the validate method to not specify yaml, in-cluster or paths together?